misc/xdg.py

```diff
@@ -2,19 +2,25 @@
 
 from gi.repository import Gio
 import sys
-import dbus
 
 def launch(desktop, *files):
     launcher = Gio.DesktopAppInfo.new_from_filename(desktop)
-    if hasattr(launcher, 'get_boolean'):
-        activatable = launcher.get_boolean('DBusActivatable')
-        if activatable:
-            bus = dbus.SessionBus()
-            service_id = launcher.get_id()
-            # cut the .desktop suffix
-            service_id = service_id[:-8]
-            bus.start_service_by_name(service_id)
-    launcher.launch(files, None)
+    try:
+        import dbus
+        if hasattr(launcher, 'get_boolean'):
+            activatable = launcher.get_boolean('DBusActivatable')
+            if activatable:
+                bus = dbus.SessionBus()
+                service_id = launcher.get_id()
+                # cut the .desktop suffix
+                service_id = service_id[:-8]
+                try:
+                    bus.start_service_by_name(service_id)
+                except dbus.DBusException:
+                    launcher.launch(files, None)
+        launcher.launch(files, None)
+    except:
+        launcher.launch(files, None)
```

Comment:
This is wrong:
 - bare `except` (without specifying exception) is a bad idea in most cases - you may handle this way something totally different that you wanted
 - in worst case, you'll launch the same action thrice

Better use something like:
```python
try:
    import dbus
    # ... try dbus, return from the function if succeed
except ImportError:
    pass
launcher.launch(files, None)
```

Reply:
Is this better?